include/wx/private/uilocale.h

```diff
@@ -33,18 +42,37 @@ class wxUILocaleImpl
     // It may return NULL in case of failure.
     static wxUILocaleImpl* CreateUserDefault();
 
-    // This function exists only for wxLocale compatibility and sets the locale
-    // corresponding to the given language.
+    // Create locale object for the given locale.
+    //
+    // It may return NULL in case of failure.
+    static wxUILocaleImpl* CreateForLocale(const wxLocaleIdent& locId);
+
+    // This function exists only for wxLocale compatibility and creates the
+    // locale corresponding to the given language.
```

Comment:
Perhaps use stronger language and say that it shouldn't be used in new and/or user code?

Reply:
It's a private header, so it could only be used inside wx itself, but I'll update the comment.